content/courses/advanced-ada/chapters/aggregates.rst

```diff
@@ -886,13 +886,140 @@ Delta Aggregates
    Complete section!
 
 
+.. _Adv_Ada_Container_Aggregates:
+
 Container Aggregates
 --------------------
 
-.. admonition:: Relevant topics
+A container aggregate is a list of elements |mdash| such as :ada:`[1, 2, 3]`
+|mdash| that we use to initialize or assign to a container. For example:
 
-   - :arm22:`Container Aggregates <4-3-5>`
+.. code:: ada run_button project=Courses.Advanced_Ada.Aggregates.Simple_Container_Aggregate
 
-.. todo::
+    pragma Ada_2022;
 
-   Complete section!
+    with Ada.Containers.Vectors;
+
+    procedure Show_Container_Aggregate is
+
+       package Float_Vectors is new
+         Ada.Containers.Vectors (Positive, Float);
+
+       V : constant Float_Vectors.Vector :=
+             [1.0, 2.0, 3.0];
+
+       pragma Unreferenced (V);
+    begin
+       null;
+    end Show_Container_Aggregate;
+
+In this example, :ada:`[1.0, 2.0, 3.0]` is a container aggregate that we use
+to initialize a vector :ada:`V`.
+
+We can specify container aggregates in three forms:
+
+    - as a null container aggregate, which indicates a container without any
+      elements and is represented by the :ada:`[]` syntax;
+
+    - as a positional container aggregate, where the elements are simply
+      listed in a sequence (such as :ada:`[1, 2]`);
+
+    - as a named container aggregate, where a key is indicated for each element
+      of the list (such as :ada:`[1 => 10, 2 => 15]`).
+
+Let's look at a complete example:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Aggregates.Simple_Container_Aggregate
+
+    pragma Ada_2022;
+
+    with Ada.Containers.Vectors;
+
+    procedure Show_Container_Aggregate is
+
+       package Float_Vectors is new
+         Ada.Containers.Vectors (Positive, Float);
+
+       --  Null container aggregate
+       Null_V  : constant Float_Vectors.Vector :=
+                   [];
+
+       --  Positional container aggregate
+       Pos_V   : constant Float_Vectors.Vector :=
+                   [1.0, 2.0, 3.0];
+
+       --  Named container aggregate
+       Named_V : constant Float_Vectors.Vector :=
+                   [1 => 1.0,
+                    2 => 2.0,
+                    3 => 3.0];
+
+       pragma Unreferenced (Null_V, Pos_V, Named_V);
+    begin
+       null;
+    end Show_Container_Aggregate;
+
+In this example, we see the three forms of container aggregates. The difference
+between positional and named container aggregates is that:
+
+    - for positional container aggregates, the vector index is implied by
+      its position; while
```

Comment:
Perhaps remove "while", or move it to a new line between the two bulleted lines. It is a separator for the two bullet points but looks like a typo.

Reply:
Thanks, I agree with this suggestion.